File: code/projects/01/clojure/src/rectangle_area.clj
;;;; rectangle_area.clj calculates the area of a rectangle.
;;;;
;;;; Input: The height and width of a rectangle.
;;;; Output: The area of that rectangle.
;;;;
;;;; Usage: clojure -m rectangle_area
;;;;
;;;; Based off of CS214 Lab01 : Dr. Adams, for CS 214, at Calvin College.
;;;; Completed by: Duncan Van Keulen 
;;;; Date: 2-6-2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns rectangle_area) ; namespace function names the program

    ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
    ;;; Function rectangleArea() computes the area of a rectangle.
    ;;; Receive: itsHeight, itsWidth, a number.
    ;;; Return: the area of the corresponding rectangle.
    ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
    (defn rectangleArea [itsHeight itsWidth]
      (* itsWidth itsHeight )  ; return H*W
    )
    
    ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
    ;;; Function -main is where execution begins
    ;;; Input: the height and width of a rectangle.
    ;;; Output: the area of that rectangle.
    ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
    (defn -main []
        (println "\nTo compute the area of a rectangle,")

        (print   " enter its height: ") (flush)
        (let
            ;; Read in the height
            [ height (read) ]
            (print " enter its width: ")
            (flush)
            (let
                ;; Read in the width
                [ width (read) ]
                ;; Calculate and output the area
                (print "The area of the rectangle is: ")
                (println (rectangleArea height width))
                (flush)
            )
        )
    )  

File: code/projects/01/java/RectangleArea.java
/* RectangleArea.java computes the area of a rectangle.
 *
 * Input: The width and height of a rectangle
 * Precondition: Width/height are not negative numbers
 * Output: The area of a rectangle
 *
 * Based off of CS214 Lab01 : Dr. Adams, for CS 214, at Calvin College.
 * Completed by: Duncan Van Keulen 
 * Date: 2-6-2020
 **********************************************************/


import java.util.Scanner;  // include class for Scanner

public class RectangleArea {

     /* function rectangleArea() computes a rectangle's area, given its height and width.
      * Parameter: height, a double
      * Parameter: width, a double
      * Precondition: height/width is not negative
      * Returns: the area of the rectangle
      */
     public static double rectangleArea(double height, double width) {
        return height * width;              // return an expression
     } // rectangleArea method
	
  // main program
  public static void main(String[] args) {
      // prompt for radius
      System.out.println("\n\nTo compute the area of a rectangle,");
      System.out.print(" enter its height: ");

      // Create a connection named keyboard to standard in
      Scanner keyboard = new Scanner(System.in);

      //Get the height from the user
      double height = keyboard.nextDouble();

      System.out.print(" enter its width: ");
      
      //Get the width number from the user
      double width = keyboard.nextDouble();

      //Close the keyboard scanner (good practice)
      keyboard.close();

      // output area
      System.out.println("\nThe area is " + rectangleArea(height, width) + "\n");
  } // main method

} // class RectangleArea